src/internal/single-tab-stop/keyboard-nav-util.tsx

```diff
@@ -0,0 +1,131 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from 'react';
+import { SingleTabStopNavigationAPI, SingleTabStopNavigationProvider } from '../../../src/internal/single-tab-stop';
+
+import handleKey from '../../../src/internal/utils/handle-key';
+import { circleIndex } from '../../../src/internal/utils/circle-index';
+import { KeyCode } from '../../../src/internal/keycode';
+import { getAllFocusables } from '../../../src/internal/focus-lock/utils';
+
+export default function SingleTabStopKeyboardNav({
+  children,
+  matchId,
+  direction = 'horizontal',
+}: {
+  children: React.ReactNode;
+  matchId: string;
+  direction?: 'horizontal' | 'vertical';
+}) {
+  const navigationAPI = React.useRef<SingleTabStopNavigationAPI>(null);
+  const containerObjectRef = React.useRef<HTMLDivElement>(null);
+  const focusedIdRef = React.useRef<null | string>(null);
+
+  function onUnregisterActive(focusableElement: HTMLElement) {
+    // Only refocus when the node is actually removed (no such ID anymore).
+    const target = navigationAPI.current?.getFocusTarget();
+
+    if (target && target.dataset.itemid !== focusableElement.dataset.itemid) {
+      target.focus();
+    }
+  }
+
+  function getNextFocusTarget(): null | HTMLElement {
+    if (containerObjectRef.current) {
+      const buttons: HTMLButtonElement[] = Array.from(containerObjectRef.current.querySelectorAll(matchId));
+      return buttons.find(button => button.dataset.itemid === focusedIdRef.current) ?? buttons[0] ?? null;
```

Comment:
This integration is rather quite implicit: the interactive elements must have the data-itemid and must be all assigned a selector. I wonder if this can be simplified. For instance, here we can find all interactive elements using getAllFocusables and then instead of matchId expose some `matchFocusTarget(focusedItemId, focusableElements): null | Element` - this way the matching can be performed on the consumer end and can use any mechanism so not relying on assigning the specific data- attribute.